classes/class-wpcom-liveblog-cron.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * Class WPCOM_Liveblog_Cron
+ *
+ * This class integrates with the WP CRON API to handle Housekeeping for the plugin
+ * It registers hooks to handle automated tasks across the plugin.
+ *
+ */
+
+class WPCOM_Liveblog_Cron {
+
+    /**
+     * Sets up the class.
+     */
+    public static function load() {
+        self::configure_autoarchive_cron();
+    }
+
+    /**
+     * Configures the CRON Entry to Check for the Auto Archive Expiry on all Live Blogs.
+     * @return mixed
+     */
+    private static function configure_autoarchive_cron() {
+
+        if ( ! wp_next_scheduled( 'auto_archive_check_hook' ) ) {
+            wp_schedule_event( strtotime('today midnight'), 'daily', 'auto_archive_check_hook');
+        }
+
+        add_action( 'auto_archive_check_hook', array( __CLASS__, 'execute_auto_archive_housekeeping' ) );
+    }
+
+    /**
+     * The method that details the housekeepng to undertake during the execution of the CRON
+     * task.
+     * @return mixed
+     */
+    public static function execute_auto_archive_housekeeping() {
+
+        //If Auto Archive is enabled,
+        if ( null !== WPCOM_Liveblog::$auto_archive_days ) {
+
+            $posts = new WP_Query( array(
+                'order'      => 'ASC',
+                'orderby'    => 'ID',
+                'meta_key'   => 'liveblog',
```

Comment:
Could we use `WPCOM_Liveblog::$auto_archive_expiry_key` here instead?
Also, how many posts will this query return - will it be enough to catch all the Liveblogs we'll need to archive?

Reply:
@philipjohn ```WPCOM_Liveblog::$auto_archive_expiry_key``` references the key on the meta table that holds the expiry date, in the query above I'm looking for ALL posts that have a liveblog metakey indicating its been activated as a Liveblog post. Then I loop them check any set expiry date and update as required.

If preferred I could just grab only those with a meta_value of 'enabled'? so we skip anything thats already archived.